functions.sh

```diff
@@ -20,6 +20,37 @@ error() { echo "$@" >&2 ; return 1 ; }
 info() { ((${VERBOSE:-0})) && echo "$@" >&2 ; return 0 ; }
 debug() { ((${VERBOSE:-0})) && echo "$@" >&2 ; return 0 ; }
 
+safeboot_dir() {
+	[[ -n $1 ]]	\
+	|| die "Internal error in caller of safeboot_dir"
+	case "$1" in
+	bin)	echo "$TOP/bin";;
```

Comment:
should there be a check to ensure that `$TOP` is set? Or is the expectation that an empty `$TOP` means it is installed in `/`?